backend/api/routers/auth.py

```
@@ -0,0 +1,279 @@
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail={
+                "code": AuthErrorCodes.DATABASE_ERROR,
+                "message": "Failed to update profile"
+            }
+        ) from e
+
+
+@router.delete("/profile", response_model=SuccessResponse[Dict[str, bool]])
+async def delete_profile(
+    current_user: UserProfile = Depends(get_current_user),
```

Ruff (0.11.9)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)